### PR TITLE
Issue 88

### PR DESCRIPTION
**Pull request by azubieta**

### CODE REVIEW

File: src/UsbMonitor_linux.cpp [hunk 1]
```diff
@@ -105,62 +126,53 @@ void UsbMonitor_linux::start()
...
+UsbMonitor_linux::~UsbMonitor_linux()
+{
+    relaseCallbacks();
+    libusb_exit(usb_ctx);
+}
 
-    if (end == std::find_if(begin, end, [fd](QSocketNotifier* sn) { return sn->socket() == fd; })) {
+void UsbMonitor_linux::createSocketMonitor(int fd)
+{
+    if (!m_fdMonitors.contains(fd)) {
```
Comment (raoulh) [inline src/UsbMonitor_linux.cpp]: please } on newline

File: src/UsbMonitor_linux.cpp [hunk 2]
```diff
@@ -105,62 +126,53 @@ void UsbMonitor_linux::start()
...
-    const auto end = std::end(monitoredFds);
-    monitoredFds.erase(std::remove_if(begin, end, [fd](QSocketNotifier* sn) { return sn->socket() == fd; }), end);
-}
-
-UsbMonitor_linux::~UsbMonitor_linux()
+void UsbMonitor_linux::releaseSocketMonitor(int fd)
 {
-    stop();
-    libusb_exit(usb_ctx);
+    if (m_fdMonitors.contains(fd)) {
```
Comment (raoulh) [inline src/UsbMonitor_linux.cpp]: please } on newline

**Comment by azubieta** [comment]
Fixed

**Comment by raoulh** [comment]
thanks

**Pull request merged**